src/qz/installer/certificate/firefox/FirefoxCertificateInstaller.java

```diff
@@ -155,4 +164,51 @@ public static void installPolicy(AppLocator app, X509Certificate cert) {
     public static boolean checkRunning(AppLocator app, boolean isSilent) {
         throw new UnsupportedOperationException();
     }
+
+    public static void restartFirefox(ArrayList<AppLocator> appList) throws IOException {
+        ArrayList<Path> processPaths = new ArrayList<>();
+        HashMap<AppLocator, Path> appsToRestart = new HashMap<>();
+
+        String fileExtention = SystemUtilities.isWindows() ? ".exe" : "";
+
+        AppAlias.Alias[] aliases = AppAlias.FIREFOX.getAliases();
+        String[] appNames = new String[aliases.length];
+        for (int i = 0; i < appNames.length; i++) {
+            appNames[i] = aliases[i].posix + fileExtention;
+        }
+
+        processPaths.addAll(TaskControl.locateProcessPaths(true,appNames));
+
+        log.warn("Found " + processPaths.toString() + " running");
+        for (AppLocator app : appList) {
+            Path appPath = Paths.get(app.getPath()).toRealPath();
+            //todo change app.getPath to return a path object?
+            for (Path processPath : processPaths) {
+                if (processPath.startsWith(appPath)) appsToRestart.put(app, processPath);
+            }
+        }
+        String text = "The following must restart for the changes to take effect.";
+        boolean shouldPrompt = false;
+
+        for (Map.Entry<AppLocator, Path> pair: appsToRestart.entrySet()) {
+            AppLocator app = pair.getKey();
+
+            if (app.getVersion().lessThan(Version.forIntegers(60))) {
+                shouldPrompt = true;
+                text += "\n" + app.getName() + " Version: " + app.getVersion();
+            } else {
+                executeRestartRequired(pair.getValue());
+            }
+        }
+
+        if (shouldPrompt) {
+            //Todo Remove this debugging log
+            log.warn(text);
+        }
+    }
+
+    public static void executeRestartRequired(Path processPath) {
```

Comment:
This function is only two lines, can be placed directly inside the loop for improved readiblity.